### ISSUE TITLE
The parameter P in the pool will not be updated during training

### ISSUE THREAD

**Issue by Allen-lz** [filipradenovic/cnnimageretrieval-pytorch#61]
Hi~
When I use the following command to train the network:
python3 -m cirtorch.examples.train YOUR_EXPORT_DIR 
            --gpu-id '0' 
            --training-dataset 'retrieval-SfM-120k' 
            --loss 'triplet' --loss-margin 0.5 
            --optimizer 'adam' 
            --lr 1e-6 
            --arch 'resnet50' 
            --pool 'gem' 
            --whitening 
            --neg-num 5 
            --query-size=2000 --pool-size=20000 
            --batch-size 5 --image-size 1024 --epochs 100 
            --test-datasets 'roxford5k,rparis6k' --test-freq 5 

I print p in the forward of the pool and find that p is always 3, its value has not changed.
Could you tell me how to solve this problem so that P will be updated during training? thank you.

**Comment by Allen-lz** [comment]
Excuse me, I have solved the problem
